mantidproject/mantid
# Change reference angle for sum in Q in RRO

*Experimental work*

When summing in Q, there was previously a workaround for the reference angle used in the projection in order to make it work with ConvertUnits for the final conversion to Q. Instead, use our own conversion to Q (using RefRoi so we can specify the correct angle). Then we can simply use the reference detector angle for the projection. We just need to ensure the same angle is used in the conversion to Q to match.

**To test:**



- Ensure your default save directory is set, and that it is also in your search directories for loading.
- Run the following script in v5.1.1:
```
Load(Filename='INTER43583', OutputWorkspace='INTER43583')
ReflectometryReductionOne(InputWorkspace='INTER43583', WavelengthMin=1.5, WavelengthMax=17.0,
                          ThetaIn=1.3, ProcessingInstructions='47-61',
                          SummationType='SumInQ', ReductionType='DivergentBeam')
SaveNexus(InputWorkspace='IvsQ_43583', Filename='IvsQ_43583_reference.nxs')
```
- Run this script in this dev version
```
Load(Filename='INTER43583', OutputWorkspace='INTER43583')
ReflectometryReductionOne(InputWorkspace='INTER43583', WavelengthMin=1.5, WavelengthMax=17.0,
                          ThetaIn=1.3, ProcessingInstructions='47-61',
                          SummationType='SumInQ', ReductionType='DivergentBeam')
#SaveNexus(InputWorkspace='IvsQ_43583', Filename='IvsQ_43583_reference.nxs')
Load(Filename='IvsQ_43583_reference.nxs', OutputWorkspace='IvsQ_43583_reference')
RebinToWorkspace('IvsQ_43583', 'IvsQ_43583_reference', OutputWorkspace='IvsQ_43583')
Divide('IvsQ_43583', 'IvsQ_43583_reference', OutputWorkspace='IvsQ_43583_difference')
```
- Plot `IvsQ_43583` and `IvsQ_43583_reference`. They should be the same except for binning.
- Plot `IvsQ_43583_difference`. It should be close to 1 and not contain any structural information/pattern.

Refs #23011



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
